src/controller/java/src/chip/devicecontroller/DeviceAttestationDelegate.java

```diff
@@ -0,0 +1,42 @@
+package chip.devicecontroller;
+
+/**
+ * Only one of the following delegate callbacks should be implemented.
+ *
+ * <p>If DeviceAttestationFailureCallback is implemented, then it will be called when device
+ * attestation fails, and the client can decide to continue or stop the commissioning.
+ *
+ * <p>If DeviceAttestationFailureCallback is implemented, then it will always be called when device
```

Comment:
It should be DeviceAttestationCompletionCallback

Reply:
Do you mean to use the pointer from the `DeviceAttestationFailureCallback` or `DeviceAttestationCompletionCallback` callback as a parameter?

Reply:
Do you mean to use the pointer from the `DeviceAttestationFailureCallback` or `DeviceAttestationCompletionCallback` callback as a parameter?